docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/jira.md

```diff
@@ -0,0 +1,659 @@
+| `updateIssue.addLabels`      | `array`  | N        | Labels to add to the issue.                                       |
+| `updateIssue.removeLabels`   | `array`  | N        | Labels to remove from the issue.                                  |
+| `updateIssue.customFields`   | `object` | N        | Custom fields to update.                                          |
+
+#### Output
+
+This step does not produce any output values.
+
+#### Example
+
+This example updates an existing issue's status and adds a comment with promotion details.
+
+```yaml
+steps:
+- uses: jira
+  config:
+    credentials:
+      secretName: jira-credentials
+    updateIssue:
+      issueKey: DEPLOY-123
+      status: "IN PROGRESS"
+      summary: "Deploy ${{ imageFrom(vars.imageRepo).Tag }} to ${{ ctx.stage }} - IN PROGRESS"
+      addLabels:
+      - deploying
+      - "${{ ctx.stage }}-promotion"
+      customFields:
+        customfield_10000: "${{ ctx.stage }} Environment"
+        customfield_10001: "${{ ctx.promotion }}"
+```
+
+### Delete Issue
+
+Deletes a Jira issue and optionally its subtasks.
+
+#### Configuration
+
+| Name                         | Type      | Required | Description                                    |
+| ---------------------------- | --------- | -------- | ---------------------------------------------- |
+| `deleteIssue.issueKey`       | `string`  | Y        | The Jira Issue Key (e.g., EXT-123).            |
+| `deleteIssue.deleteSubtasks` | `boolean` | N        | If true, all subtasks will be deleted as well. |
+
+#### Output
+
+This step does not produce any output values.
+
+#### Example
+
+This example deletes a Jira issue and all its subtasks when a promotion fails.
+
+```yaml
+steps:
+# existing steps create issue and other promotion steps
+# ....
+# ....
+# on failure cleanup logic
+- as: on-failure-cleanup-issue
+  uses: jira
+  if: ${{ failure() }}
+  config:
+    credentials:
+      secretName: jira-credentials
+    deleteIssue:
+      issueKey: "${{ freightMetadata(ctx.targetFreight.name, 'jira-issue-key') }}"
+      deleteSubtasks: true
+```
+
+### Search Issues
+
+Searches for Jira issues using JQL ([Jira Query Language](https://support.atlassian.com/jira-service-management-cloud/docs/use-advanced-search-with-jira-query-language-jql/)).
+
+#### Configuration
+
+| Name                         | Type      | Required | Description                                                                                   |
+| ---------------------------- | --------- | -------- | --------------------------------------------------------------------------------------------- |
+| `searchIssue.jql`            | `string`  | Y        | The JQL query to search for issues.                                                           |
+| `searchIssue.expectMultiple` | `boolean` | N        | If true, expects multiple results and returns the first matching result. If false, expects single result and fails with >1 results. |
```

Comment:
Just a note that this is why I've _completely_ given up on trying to make tables look pretty in the source. Change one entry and nothing lines up anymore. 🤷‍♂️ 

No action needed.